encfs/BlockFileIO.cpp

```diff
@@ -171,11 +171,11 @@ ssize_t BlockFileIO::read(const IORequest &req) const {
   return result;
 }
 
-bool BlockFileIO::write(const IORequest &req) {
+int BlockFileIO::write(const IORequest &req) {
   CHECK(_blockSize != 0);
 
   off_t fileSize = getSize();
-  if (fileSize < 0) return false;
+  if (fileSize < 0) return fileSize;
```

Comment:
This changes the boolean check.  Normally, it would return false if `fileSize < 0` so line 256 would break.  However you now evaluate the negative `int` in the conditional so whether it returns a positivie or negative it will be evaulated as `true` and execute break.

Is this intended to change the behavior?